### PR TITLE
Rnn forward logic test

### PR DESCRIPTION
**Pull request by Superjom**
rnn logic test by python is ready.

### CODE REVIEW

File: paddle/framework/operator.h [hunk 1]
```diff
@@ -174,7 +174,10 @@ class OperatorContext {
   template <typename T>
   T* Output(const size_t index) const {
     auto var = OutputVar(index);
-    PADDLE_ENFORCE(var != nullptr, "Output(%d) should not be nullptr", index);
+    PADDLE_ENFORCE(var != nullptr,
+                   "Output(%d) not be nullptr, which means variable [%s] not "
```
Comment (qingqing01) [inline paddle/framework/operator.h]: not - > does not

File: paddle/operators/recurrent_op.cc [hunk 2]
```diff
@@ -67,22 +69,31 @@ void RecurrentAlgorithm::Run(const Scope& scope,
...
+  // TODO(superjom) Only two scopes are needed for inference, this case will be
   // supported later.
-  auto step_scopes =
-      scope.FindVar(arg_->step_scopes)->GetMutable<std::vector<Scope*>>();
+  auto step_scopes_var = scope.FindVar(arg_->step_scopes);
+  PADDLE_ENFORCE(step_scopes_var != nullptr, "");
+  auto step_scopes = step_scopes_var->GetMutable<std::vector<Scope*>>();
+
+  // Now all variables in scope must be created outside of op.
+  auto var = scope.FindVar(arg_->step_net);
```
Comment (qingqing01) [inline paddle/operators/recurrent_op.cc]: var -> net_var

File: paddle/operators/recurrent_op.cc [hunk 3]
```diff
@@ -67,22 +69,31 @@ void RecurrentAlgorithm::Run(const Scope& scope,
...
+  PADDLE_ENFORCE(var != nullptr, "no stepnet called %s in scope",
+                 arg_->step_net);
```
Comment (qingqing01) [inline paddle/operators/recurrent_op.cc]: If check step_net in CreateScopes, please remove its check in RecurrentAlgorithm::InferShape() fuction.

Comment (qingqing01) [inline paddle/operators/recurrent_op.cc]: Maybe we need `PADDLE_ENFORCE_NOTNULL` macro?

File: python/paddle/v2/framework/tests/test_recurrent_op.py [hunk 4]
```diff
@@ -31,58 +31,65 @@ class TestRNN(unittest.TestCase):
...
-        stepnet.complete_add_op(True)
 
+        self.scope = core.Scope()
+
+        self.create_global_variables()
+        self.create_step_net()
+        rnn_op = self.create_rnn_op()
+
+        ctx = core.DeviceContext.create(core.CPUPlace())
+        print 'infer_shape'
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_recurrent_op.py]: remove this line

File: paddle/operators/recurrent_op.cc (hunk 3 above)
Comment (Superjom) [inline paddle/operators/recurrent_op.cc]: another enforce is not expensive, no need to remove.

**Pull request merged**